lib/Charge.php

```diff
@@ -118,9 +118,19 @@ class Charge extends ApiResource
      * @param array|string|null $options
      *
      * @return Charge The refunded charge.
+     *
+     * @deprecated This method is deprecated and planned for removal. Prefer using
+     *             \Stripe\Refund::create() instead.
+     * @see \Stripe\Refund::create()
      */
+    // TODO: remove in next major version
     public function refund($params = null, $options = null)
     {
+        trigger_error(
+            "This method is deprecated and planned for removal. Prefer using \Stripe\Refund::create() instead",
+            E_USER_WARNING
+        );
```

Comment:
This I'm quite meh about. While devs should have warning disabled, I am not convinced everyone would and this is quite dangerous to do to users on a minor rev.

Reply:
I've just found out that `E_USER_DEPRECATED` exists since PHP 5.3, so I've updated the PR to use it instead of `E_USER_WARNING`.

Reply:
I'm still quite bearish here. I don't think it helps to surface this client-side ultimately. Like sure people will see it and maybe decide to do something with it, but the risk of having this appear to a real customer seems not worth it